README.md

```diff
@@ -1,28 +1,95 @@
 # StableHLO
 
-StableHLO is an operation set that expresses ML computations. It has been
-originally bootstrapped from
-[the MHLO dialect](https://github.com/tensorflow/mlir-hlo#meta-hlo-dialect-mhlo)
-and enhances it with additional functionality, including serialization and
-versioning.
+StableHLO is an operation set for high-level operations (HLO) in machine
+learning (ML) models. Essentially, it's a portability layer between different
+ML frameworks and ML compilers: ML frameworks that produce StableHLO programs
+are compatible with ML compilers that consume StableHLO programs.
 
-StableHLO is a portability layer between ML frameworks and ML compilers.
-We are aiming for adoption by a wide variety of ML frameworks including
-TensorFlow, JAX and PyTorch, as well as ML compilers including XLA and IREE.
+Our goal is to simplify and accelerate ML development by creating more
+interoperability between various ML frameworks (such as TensorFlow, JAX and
+PyTorch) and ML compilers (such as XLA and IREE).
 
-## Development
+StableHLO is based on the MHLO dialect and enhances it with additional
+functionality, including serialization and versioning. This provides
+[backward and forward
+compatibility](https://github.com/openxla/stablehlo/blob/main/rfcs/20220912-compatibility.md)
+guarantees for StableHLO programs and ensures compatibility between frameworks
+and compilers, even as StableHLO continues to evolve.
+
+This repository includes the [StableHLO
+specification](https://github.com/openxla/stablehlo/blob/main/docs/spec.md)
+along with implementations in C++ and Python, which you can use to build
+StableHLO programs.
+
+## Build steps
+
+  Here's how to build the StableHLO repo:
+
+1. First clone this repo and the LLVM/MLIR git repository:
+
+   ```sh
+   git clone https://github.com/openxla/stablehlo
+   ```
+
+   ```sh
+   cd stablehlo && git clone https://github.com/llvm/llvm-project.git
+   ```
+
+2. Make sure you check out the correct commit in the LLVM repository:
+
+   ```sh
+   (cd llvm-project && git fetch && git checkout $(cat ../build_tools/llvm_version.txt))
+   ```
+
+   You need to do this every time you run git pull in the stablehlo repo
+   (in case `llvm_version.txt` changes).
```

Comment:
Since we're here, how about we simplify this a bit to say "You need to do this every time `llvm_version.txt` changes"? This will avoid the need to talk about git pull. Also, llvm_version.txt can change due to other reasons.

Reply:
SGTM